### DISCUSSION TITLE
Getting double `pre` and `code` elements

### DISCUSSION THREAD

**Discussion by babakfp** [markedjs/marked#2659] [Q&A] [answered]
Hi

I'm getting this output by using Shiki and this code https://github.com/markedjs/marked/issues/2122#issuecomment-871735769 is there a way to tell marked not to create its `pre` and `code` tags?

![Screenshot 2022-11-18 010749](https://user-images.githubusercontent.com/44144724/202664406-e7a1157f-1834-493c-aeb7-e57c2ffa9af4.png)

**Comment by UziTech** [comment, marked as answer]
You can change the code renderer to not always wrap the code in pre and code tags. This code seems to work:

```js
import { marked } from 'marked';
import shiki from 'shiki';

function markdownToHtml(markdown) {
  return new Promise((resolve, reject) => {
    const markedOptions = {
      highlight: (code, lang, callback) => {
        shiki
          .getHighlighter({ theme: 'nord' })
          .then(highlighter => {
            const result = highlighter.codeToHtml(code, lang);
            callback(null, result);
          })
          .catch(error => callback(error));
      },
      renderer: {
        code(code) {
          if (code.startsWith('<pre')) {
            return code;
          }

          return false;
        }
      }
    };

    const markedCallback = (err, html) => {
      if (err) return reject(err);
      resolve(html);
    };

    marked.use(markedOptions);

    marked(markdown, markedCallback);
  });
}

console.log(await markdownToHtml("```js\nconsole.log('test');\n```\n"));
```

**Reply by babakfp** [reply]
I think adding the `renderer: {` rewrites the whole default code for other elements like heading and etc. And because of that, it throws this error:
```
renderer.text is not a function
```
It throws me this error when I have an empty  `renderer: {` object and a basic `# Hello, World!` heading to render.

**Reply by UziTech** [reply]
That is why I'm my code sample I use `marked.use` to set the options instead of sending them in the marked call. `marked.use` merges the renderer intermittently.

**Reply by laygir** [reply]
I really need this and somehow can't make it work, will try to get back with a reproduction case. But fwiw, this should (regardless shiki) still work, right?

**Reply by UziTech** [reply]
You should use [marked-highlight](https://www.npmjs.com/package/marked-highlight) to make highlighting code easy in marked.
